pkg/pool-linear/contracts/LinearPool.sol

```diff
@@ -46,36 +55,53 @@ contract LinearPool is BasePool, IGeneralPool, LinearMath, IRateProvider {
 
     uint256 private constant _TOTAL_TOKENS = 3; // Main token, wrapped token, BPT
 
-    // Linear Pools don't lock any BPT, since they fully support having zero main and wrapped token balances
-    uint256 private constant _LINEAR_MINIMUM_BPT = 0;
-    uint256 private constant _MAX_TOKEN_BALANCE = 2**(112) - 1;
+    // This is the maximum token amount the Vault can hold. In regular operation, the total BPT supply remains constant
+    // and equal to _INITIAL_BPT_SUPPLY, but most of it remains in the Pool, waiting to be exchanged for tokens. The
+    // actual amount of BPT in circulation is the total supply minus the amount held by the Pool, and is known as the
+    // 'virtual supply'.
+    // The only instance in which the total supply might change is if the emergency pause is turned on, enabling an
+    // alternative way to exit the Pool which involves burning BPT. As this is expected to not happen, we optimize for
+    // success by using _INITIAL_BPT_SUPPLY instead of totalSupply(), saving a storage read. This optimization is only
+    // valid if the Pool is never paused: in the case of an emergency that leads to burned tokens, the Pool should not
+    // be unpaused and continue to be used via regular means.
+    uint256 private constant _INITIAL_BPT_SUPPLY = 2**(112) - 1;
 
     IERC20 private immutable _mainToken;
     IERC20 private immutable _wrappedToken;
 
+    // The indices of each token when registered, which can then be used to access the balances array.
     uint256 private immutable _bptIndex;
     uint256 private immutable _mainIndex;
     uint256 private immutable _wrappedIndex;
 
+    // Both BPT and the main token have a regular, constant scaling factor (equal to FixedPoint.ONE for BPT, and
+    // dependent on the number of decimals for the main token). The wrapped token's scaling factor however is two-fold:
+    // it is composed of both the decimal scaling factor, as well as an externally provided rate that is used to
+    // convert wrapped tokens to an equivalent main token amount. This external rate is expected to be ever increasing,
+    // reflecting the fact that the wrapped token appreciates in value over time (because e.g. it is accruing interest).
```

Comment:
```suggestion
    // dependent on the number of decimals for the main token). However, the wrapped token's scaling factor has two components:
    // the usual token decimal scaling factor, and an externally provided rate used to
    // convert wrapped tokens to an equivalent main token amount. This external rate is expected to be ever increasing,
    // reflecting the fact that the wrapped token appreciates in value over time (e.g.; because it is accruing interest).
```